### PR TITLE
[BIT] paddle.multiplex paddle.nn.functional.adaptive_log_softmax_with_loss

### PR DESCRIPTION
**Pull request by cszdrg**
## paddle.nn.functional.adaptive_log_softmax_with_loss
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/adaptive_log_softmax_with_loss_cn.html
a) 该api计算自适应 logsoftmax 结果以及 input 和 label 之间的负对数似然 **torch中无对应**
b) 手写rule进行计算
c）发现参数**list[list[Tensor([16,8],"float32"),Tensor([8,5],"float32"),],list[Tensor([16,4],"float32"),Tensor([4,5],"float32"),],list[Tensor([16, 2],"float32"),Tensor([2, 5],"float32"),],]**
初始化错误，查询base相关代码发现如果list里面包list再包tensor便无法初始化
d）仿照_handle_list_or_tuple逻辑增加_handle_list_or_tuple_torch函数，处理list中嵌套list的情况
e）解决后进行测试，pass

## paddle.multiplex
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/multiplex_cn.html
a） 该api根据给定的 index 参数，从每个输入 Tensor 中选择特定行构造输出 Tensor。**torch中无对应**
b）手写rule

### CODE REVIEW

**Review by yuwu46 — APPROVED** [review]

**Review by cangtianhuang — APPROVED** [review]
LGTM
